docs/reference/rockcraft.yaml.rst

```diff
@@ -84,10 +84,14 @@ which is typically used with static binaries or
    The notation "ubuntu:<channel>" is also supported for some channels, but this
    format is deprecated and should be avoided.
 
+.. note::
+   Base ``ubuntu@24.04`` is still unstable and under active development. To use
+   it, ``build-base`` *must* be ``devel``.
+
 ``build-base``
 --------------
 
-**Type**: One of ``ubuntu@20.04 | ubuntu@22.04``
+**Type**: One of ``ubuntu@20.04 | ubuntu@22.04 | devel``
```

Comment:
I think we've discussed it before, but my concern remains here: even though `devel` is quite explicit and resulting rocks should be marked as so, what will happen is that, especially for continuous pipelines, developers will need to manually change this parameter whenever 24.04 is released. 

That time window is quite narrow.

Between 24.04 release and the developer changing this recipe, rocks might be built and released, and since we don't support package pinning, this can result in a catastrophic integrity problem. Eg.: the rock `python:3.11-24.04_edge` might be shipped with `python3.12` once 24.04 is released, cause `devel` will suddenly become 24.10.

Am I looking at this wrong?

Reply:
Ah I see what you mean. You are right: let's say that we release Rockcraft 1.2.0 with this change. In May, an existing rockcraft.yaml project with `build-base: devel` will suddenly start using 24.10 base even though the version of Rockcraft itself hasn't changed.

The solution might be to [do what Snapcraft did](https://github.com/snapcore/snapcraft/pull/4532) and specifically use the daily Noble images for `devel`, instead of the latest "rolling" stream. Then when we decide to start supporting 24.10 as "devel" we'll have to explicitly release a new Rockcraft version with the change.

Reply:
note: The reason that I didn't adopt that Snapcraft change in this PR is that apparently there is/was a problem with the remote containing the daily images. Looks like that's been/will soon be solved, though

Reply:
Yes I think that approach would be slightly better, even though people tend to use `latest` version in their ci pipelines. Nonetheless, the use of latest is always an argument we can fight against, so ...y, you solution is an improvement

Reply:
This should not be a problem, anyone tracking devel should be in that spirit, tracking the latest development.

Once 24.04 is release, devel indeed goes to 24.10, because it is devel, the timewindow is short now due to the nature of what we are thinking about, but if you change that to think about 26.04, the scale does make more sense.

Reply:
Note: our agreement here during last week's sync was to follow Snapcraft's notion of `devel`, in that there is no expectation of stability because even for a specific target (like noble) things change very frequently up until stabilization. I'll add Snapcraft's warning message as agreed

Reply:
agreed. all good on my side.